daft/expressions/expressions.py

```diff
@@ -2359,6 +2359,41 @@ def day_of_week(self) -> Expression:
         """
         return Expression._from_pyexpr(native.dt_day_of_week(self._expr))
 
+    def day_of_year(self) -> Expression:
```

Comment:
```suggestion
    def dayofyear(self) -> Expression:
```

Let's keep the name the same across SQL and Expression API. Either or works for me but right now we're mix-matching.

```python
dayofweek
dayofyear

day_of_week
day_of_year
```

I know spark and duckdb have "dayofweek" and "dayofyear" already

Reply:
so this is consistent with our `dayofweek` sql implementation. I know they diverge. But we initially did that because of the precedent set in sql. Generally I do prefer the words to be `_` delimited (eg. `day_of_week` instead of `dayofweek`).

Reply:
Yes, I understand _why_ they are this way, but I am inviting you to make a better experience for daft. If you prefer them to be delimited (as do I) then let's do it. 

If you have concerns about backwards-compatibility for our existing "dayofweek" then add the "day_of_week" alias and it's your decision if you want to deprecate it or not in 0.5.0.

Reply:
The other option is make the python API `.dayofweek` and `.dayofyear` even though we don't personally like it, but it's consistent with the wider ecosystem. Just trying to reduce the inconsistencies we are introducing to our APIs.

Reply:
didnt want to add a test for a deprecated function, but added a warning for it if one tries to call `dayofweek`


<img width="764" alt="image" src="https://github.com/user-attachments/assets/86cadfc7-8a5c-4a21-88a5-d9d5b504f1cd" />